### PR TITLE
Bump Rust Toolchain

### LINKED ISSUE #9362: Bump Rust Toolchain
Our Rust toolchain nightly version is more than a year old, we need to replace it with a current one.

This includes necessary code fixes / lints. Optionally, better idioms.

### PR DESCRIPTION
**Pull request by mwu-tow**
### Pull Request Description
This PR updates the Rust toolchain to recent nightly.

Most of the changes are related to fixing newly added warnings and adjusting the feature flags. Also the formatter changed its behavior slightly, causing some whitespace changes.

Other points:
* Changed debug level of the `buildscript` profile to `lint-tables-only` — this should improve the build times and space usage somewhat.
* Moved lint configuration to the worksppace `Cargo.toml` definition. Adjusted the formatter appropriately.
* Removed auto-generated IntelliJ run configurations, as they are not useful anymore.
* Added a few trivial stdlib nightly functions that were removed to our codebase.
* Bumped many dependencies but still not all:
  * `clap` bump encountered https://github.com/clap-rs/clap/issues/5407 — for now the warnings were silenced by the lint config.
  * `octocrab` — our forked diverged to far with the original, needs more refactoring.
  * `derivative` — is unmaintained and has no updated version, despite introducing warnings in the generated code. There is no direct replacement.

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [ ] The documentation has been updated, if necessary.
- [ ] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [ ] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [ ] Unit tests have been written where possible.
  - [ ] If GUI codebase was changed, the GUI was tested when built using `./run ide build`.

### CODE REVIEW

File: build/base/src/extensions/option.rs [hunk 1]
```diff
@@ -5,6 +5,18 @@
...
 
-impl<T> OptionExt<T> for Option<T> {}
+impl<T> OptionExt<T> for Option<T> {
+    fn contains<U>(&self, x: &U) -> bool
+    where U: PartialEq<T> {
+        match self {
+            Some(y) => x.eq(y),
+            None => false,
+        }
+    }
```
Comment (farmaazon) [inline build/base/src/extensions/option.rs:21]: lints some ago recommended `is_some_and`.

File: build/base/src/extensions/result.rs [hunk 2]
```diff
@@ -119,4 +123,12 @@ impl<T, E> ResultExt<T, E> for std::result::Result<T, E> {
...
         }
     }
+
+    fn contains<U>(&self, x: &U) -> bool
+    where U: PartialEq<T> {
+        match self {
+            Ok(y) => x == y,
+            Err(_) => false,
+        }
+    }
```
Comment (farmaazon) [inline build/base/src/extensions/result.rs:133]: `is_ok_and`

BTW those two functions were a reason of removing `contains`.

File: build/ci_utils/src/actions/artifacts/raw.rs [hunk 3]
```diff
@@ -108,7 +106,7 @@ pub mod endpoints {
         serde_json::from_value(body).anyhow_err()
     }
 
-    #[context("Failed to finalize upload of the artifact `{}`.", artifact_name.as_ref())]
+    // #[context("Failed to finalize upload of the artifact `{}`.", artifact_name.as_ref())]
```
Comment (farmaazon) [inline build/ci_utils/src/actions/artifacts/raw.rs]: commented code.

File: build/ci_utils/src/archive/extract_files.rs [hunk 4]
```diff
@@ -9,5 +9,6 @@ pub trait ExtractFiles {
     /// IMPORTANT: If the function uses its input path to generate an output path, care must be
     /// taken that the output path is not in an unexpected location, especially if coming from an
     /// untrusted archive.
+    #[allow(async_fn_in_trait)]
```
Comment (farmaazon) [inline build/ci_utils/src/archive/extract_files.rs]: If we allow something, there's should be a comment explaining why lint is not right here.

**Review by farmaazon — APPROVED** [review]

**Review by 4e6 — APPROVED** [review]

File: build/base/src/extensions/option.rs (hunk 1 above)
Comment (mwu-tow) [reply build/base/src/extensions/option.rs:21]: I find `is_some_and(|x| x == target)` less readable and somewhat annoying. What's worse, it takes ownership of the inner value, so it is not really a replacement in many cases. I think the recommendation was using `iter().any(…)` but I'd rather keep `contains`.

**Pull request merged**